### PR TITLE
[RTL] Expose missing default properties, ensure bbcode is reparserd when these are changed.

### LINKED ISSUE #7491: Add `horizontal_alignment` property for RichTextLabel
### Describe the project you are working on

UI

### Describe the problem or limitation you are having in your project

This is just a little/minor detail. You can do `[center]` for horizontal alignment on `RichTextLabel`, but when you're assigning the text of that `RichTextLabel` programatically you may "accidentally" wipe that `[center]` (e.g. forget `+=` instead of `=`, or `append_text`).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A `rich_text_label.horizontal_alignment` property which avoids the issue of wiping `[center]` and eliminates its need.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```gdscript
rich_text_label.horizontal_alignment = CENTER
```

Instead of:

```gdscript
rich_text_label.text = "[center]"
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Y

### Is there a reason why this should be core and not an add-on in the asset library?

GDScript API

### PR DESCRIPTION
**Pull request by bruvzg**
For whatever reason, default alignment and some other properties were not accessible. As well as not fully applied to the bbcode and not fully updated when changed.

- *Production edit: This closes https://github.com/godotengine/godot-proposals/issues/7491.*

### CODE REVIEW

File: scene/gui/rich_text_label.cpp [hunk 1]
```diff
@@ -5726,19 +5726,89 @@ void RichTextLabel::set_text_direction(Control::TextDirection p_text_direction)
 
 	if (text_direction != p_text_direction) {
 		text_direction = p_text_direction;
-		main->first_invalid_line.store(0); //invalidate ALL
-		_validate_line_caches();
+		if (!text.is_empty()) {
+			_apply_translation();
+		} else {
+			main->first_invalid_line.store(0); //invalidate ALL
```
Comment (Geometror) [inline scene/gui/rich_text_label.cpp:5732]:
```suggestion
			main->first_invalid_line.store(0); // Invalidate all lines.
```
Same for all other occurrences.

**Review by Calinou — APPROVED** [review]
Tested locally (rebased on top of `master` https://github.com/godotengine/godot/commit/607b230ffe120b2757c56bd3d52a7a0d4e502cfe), it works as expected.

**Testing project:** [test-richtextlabel-default-align.zip](https://github.com/user-attachments/files/16401879/test-richtextlabel-default-align.zip)

| BBCode surrounding the whole text on the left, default horizontal alignment properties on the right |
|:-:|
| image |

This is a more general comment that would need to be addressed in a separate PR for nodes such as Label as well, but the inspector is now quite long when all sections are collapsed. It might be worth moving **Justification Flags** (and perhaps **Tab Size** and **Tab Stops**) to its own section that could be collapsed (since it doesn't need to be tweaked often):

image

File: doc/classes/RichTextLabel.xml [hunk 2]
```diff
@@ -625,6 +625,12 @@
 		<member name="hint_underlined" type="bool" setter="set_hint_underline" getter="is_hint_underlined" default="true">
 			If [code]true[/code], the label underlines hint tags such as [code skip-lint][hint=description]{text}[/hint][/code].
 		</member>
+		<member name="horizontal_alignment" type="int" setter="set_horizontal_alignment" getter="get_horizontal_alignment" enum="HorizontalAlignment" default="0">
+			Controls the text's horizontal alignment. Supports left, center, right, and fill, or justify. Set it to one of the [enum HorizontalAlignment] constants.
+		</member>
+		<member name="justification_flags" type="int" setter="set_justification_flags" getter="get_justification_flags" enum="TextServer.JustificationFlag" is_bitfield="true" default="163">
+			Line fill alignment rules. For more info see [enum TextServer.JustificationFlag].
```
Comment (Calinou) [inline doc/classes/RichTextLabel.xml:632]:
```suggestion
			Line fill alignment rules. See [enum TextServer.JustificationFlag] for more information.
```

**Comment by akien-mga** [comment]
@bruvzg Couple review comments to handle, then it should be good to merge.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**